homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -43,53 +44,49 @@ class BMWSensorEntityDescription(SensorEntityDescription):
     is_available: Callable[[MyBMWVehicle], bool] = lambda v: v.is_lsc_enabled
 
 
+TIRES = ["front_left", "front_right", "rear_left", "rear_right"]
+
 SENSOR_TYPES: list[BMWSensorEntityDescription] = [
     BMWSensorEntityDescription(
-        key="ac_current_limit",
+        key="charging_profile.ac_current_limit",
```

Comment:
Changing the key, means also changing the unique id (see https://github.com/home-assistant/core/pull/120949/files#diff-58dcc591bbd6b0e56a5d7f1b6b1a4d38486cba28db575070dac071668ba9919aR220) therefore we need a migration for it. If you still want to change the key, you should please do it in a separate PR

Reply:
That is why the [`update_unique_id()`](https://github.com/home-assistant/core/pull/120949/files#diff-d9ed91adf5c8ce40001603047a42a975ed0f7640a63d6af98b93d00784f56322) got extended (and the migration tests adjusted).

We cannot do this in a separate PR as migrating the keys before changing [`_handle_coordinator_update()`](https://github.com/home-assistant/core/pull/120949/files#diff-58dcc591bbd6b0e56a5d7f1b6b1a4d38486cba28db575070dac071668ba9919aL210-R233) will cause failures.

Reply:
@edenhaus just thought about it again, would you prefer 1 PR for migrating the keys and the update function and a second PR for adding the tire sensors?

Reply:
Created #121380 and will rebase this as soon as the other one is merged.

Reply:
Rebased after unique_ids are migrated with another PR.